src/process.cr

```diff
@@ -5,45 +5,87 @@ require "c/sys/wait"
 require "c/unistd"
 
 class Process
+  # Terminate the current process immediately. All open files, pipes and sockets
+  # are flushed and closed, all child processes are inherited by PID 1. This does
+  # not run any handlers registered with `at_exit`, use `::exit` for that.
+  #
+  # *status* is the exit status of the current process.
   def self.exit(status = 0)
```

Comment:
Aparte:

- what about renaming it to `Process.exit!` like in Ruby, to mark the difference and avoid a confusion?
- `::exit(bool)` would be nice sugar too.

If those are interesting I'll propose a follow-up PR.